data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateRandao.java

```diff
@@ -0,0 +1,107 @@
+ */
+
+package tech.pegasys.teku.beaconrestapi.handlers.v1.beacon;
+
+import static tech.pegasys.teku.beaconrestapi.BeaconRestApiTypes.EPOCH_PARAMETER;
+import static tech.pegasys.teku.beaconrestapi.BeaconRestApiTypes.PARAMETER_STATE_ID;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_BEACON;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.BYTES32_TYPE;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.Optional;
+import java.util.function.Function;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiEndpoint;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
+
+public class GetStateRandao extends RestApiEndpoint {
+  public static final String ROUTE = "/eth/v1/beacon/states/{state_id}/randao";
+  private final ChainDataProvider chainDataProvider;
+
+  public GetStateRandao(final DataProvider dataProvider) {
+    this(dataProvider.getChainDataProvider());
+  }
+
+  GetStateRandao(final ChainDataProvider chainDataProvider) {
+    super(createMetadata());
+    this.chainDataProvider = chainDataProvider;
+  }
+
+  private static EndpointMetadata createMetadata() {
+
+    final SerializableTypeDefinition<Bytes32> randaoType =
+        SerializableTypeDefinition.object(Bytes32.class)
+            .name("Randao")
+            .withField("randao", BYTES32_TYPE, Function.identity())
+            .build();
+
+    final SerializableTypeDefinition<Bytes32> responseType =
+        SerializableTypeDefinition.object(Bytes32.class)
+            .name("GetStateRandaoResponse")
+            .description("RANDAO mix for state with given 'stateId'.")
+            .withField("data", randaoType, Function.identity())
+            .build();
+    return EndpointMetadata.get(ROUTE)
+        .operationId("getStateRandao")
+        .summary("Get chain genesis details")
+        .description(
+            "Fetch the RANDAO mix for the requested epoch from the state identified by `state_id`.\n\n"
+                + "If an epoch is not specified then the RANDAO mix for the state's current epoch will be returned.\n\n"
+                + "By adjusting the `state_id` parameter you can query for any historic value of the RANDAO mix. "
```

Comment:
> By adjusting the `state_id` parameter you can query for any historic value of the RANDAO mix.
Do we want to clarify that it might not be available if the node does not have historical states?

Reply:
probably not, we'd have to say that for basically every `/beacon/states` endpoint...

Reply:
this is basically the description straight from the standard api...

Reply:
Maybe on the api doc we can mention that 404 might also mean that the BN does not have the historical state they are looking for?

Reply:
We could do, but generally 404 is the route was not found, so i'm not really fussed to change the 404 response to have a custom message unless someone feel strongly about it...